src/key.rs

```diff
@@ -218,7 +203,12 @@ impl SecretKey {
 impl ::serde::Serialize for SecretKey {
     fn serialize<S: ::serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
         if s.is_human_readable() {
-            s.collect_str(self)
+            #[cfg(any(feature = "std", feature = "alloc"))] {
+                #[allow(deprecated)]
+                s.serialize_str(&self.format_secret())
+            }
+            #[cfg(not(any(feature = "std", feature = "alloc")))]
+            s.serialize_bytes(&self[..])
```

Comment:
idk if we really want that.
serializing differently depending on features can cause serious breakage.
we can also easily implement stack based str here

Reply:
Hm, can you elaborate how to do that?

Reply:
See the last 2 commits here: https://github.com/elichai/rust-secp256k1/tree/pr-312 (feel free to cherry-pick them)

Reply:
@elichai cherry-picked, thanks!